packages/dd-trace/src/opentelemetry/context_manager.js

```diff
@@ -2,61 +2,39 @@
 
 const { AsyncLocalStorage } = require('async_hooks')
 const { trace, ROOT_CONTEXT } = require('@opentelemetry/api')
+const DataDogSpanContext = require('../opentracing/span_context')
 
 const SpanContext = require('./span_context')
 const tracer = require('../../')
 
-// Horrible hack to acquire the otherwise inaccessible SPAN_KEY so we can redirect it...
-// This is used for getting the current span context in OpenTelemetry, but the SPAN_KEY value is
-// not exposed as it's meant to be read-only from outside the module. We want to hijack this logic
-// so we can instead get the span context from the datadog context manager instead.
-let SPAN_KEY
-trace.getSpan({
-  getValue (key) {
-    SPAN_KEY = key
-  }
-})
-
-// Whenever a value is acquired from the context map we should mostly delegate to the real getter,
-// but when accessing the current span we should hijack that access to instead provide a fake span
-// which we can use to get an OTel span context wrapping the datadog active scope span context.
-function wrappedGetValue (target) {
-  return (key) => {
-    if (key === SPAN_KEY) {
-      return {
-        spanContext () {
-          const activeSpan = tracer.scope().active()
-          const context = activeSpan && activeSpan.context()
-          return new SpanContext(context)
-        }
-      }
-    }
-    return target.getValue(key)
-  }
-}
-
 class ContextManager {
   constructor () {
     this._store = new AsyncLocalStorage()
   }
 
   active () {
-    const active = this._store.getStore() || ROOT_CONTEXT
+    const activeSpan = tracer.scope().active()
 
-    return new Proxy(active, {
-      get (target, key) {
-        return key === 'getValue' ? wrappedGetValue(target) : target[key]
-      }
-    })
+    const context = (activeSpan && activeSpan.context()) || this._store.getStore() || ROOT_CONTEXT
+    if (context instanceof DataDogSpanContext) {
+      const newSpanContext = new SpanContext(context)
+      return trace.setSpanContext(ROOT_CONTEXT, newSpanContext)
+    }
```

Comment:
Wouldn't this mean it'd _never_ get the context from the OTel context manager as long as _our own_ context manager has a span of its own? When would it ever _not_ enter this branch? 🤔

Reply:
there's quite a few test cases where it wouldn't get a context from our context manager:

`active () {
    const activeSpan = tracer.scope().active()
    const context = (activeSpan && activeSpan.context()) || this._store.getStore() || ROOT_CONTEXT
    const contextSource = activeSpan ? 'activeSpan' : this._store.getStore() ? '_store' : 'ROOT_CONTEXT'
    console.log('Context is coming from:', contextSource)
    if (context instanceof DataDogSpanContext) {
      console.log('inside if statement', 'does it have ._ddContext?', context._ddContext)
      const newSpanContext = new SpanContext(context)
      return trace.setSpanContext(ROOT_CONTEXT, newSpanContext)
    }
    return context
  }`
  
  if you run the unit and integration tests with this it better illustrates it, but it's definitely possible since the with code:
  
  `with (context, fn, thisArg, ...args) {
    const span = trace.getSpan(context)
    const ddScope = tracer.scope()
    const run = () => {
      const cb = thisArg == null ? fn : fn.bind(thisArg)
      return this._store.run(context, cb, ...args)
    }
    if (span && span._ddSpan) {
      return ddScope.activate(span._ddSpan, run)
    }
    return run()
  }`
  can run a context outside the ddScope